File: examples-sdk/python/028-basic-sse-hub-streaming.py
#!/usr/bin/env python3
"""028-basic-sse-hub-streaming — Python SDK equivalent
Compile: vil compile --from python --input 028-basic-sse-hub-streaming.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

server = VilServer("sse-hub-demo", port=8080)
events = server.service_process("events")
events.endpoint("POST", "/publish", "publish")
events.endpoint("GET", "/stream", "stream")
events.endpoint("GET", "/stats", "stats")
server.compile()

File: examples-sdk/typescript/028-basic-sse-hub-streaming.ts
#!/usr/bin/env tsx
// 028-basic-sse-hub-streaming — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 028-basic-sse-hub-streaming.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const server = new VilServer("sse-hub-demo", 8080);
const events = new ServiceProcess("events");
events.endpoint("POST", "/publish", "publish");
events.endpoint("GET", "/stream", "stream");
events.endpoint("GET", "/stats", "stats");
server.service(events);
server.compile();
